### PR TITLE
Httpd tests

### PR DESCRIPTION
**Pull request by vorner**
More tests will come in future pull requests.

There are some shortcuts and such, some of them are not exactly nice, but I hope this is not such a problem in test code.

If someone knows what better to do about the cmake thing, I'm all ears.

### CODE REVIEW

File: lib/WUI/http/httpd_post.c [hunk 1]
```diff
@@ -354,7 +354,7 @@ void httpd_post_finished(void *connection,
     if (connection != NULL && status_code == 200) {
         strncpy(response_uri, endpoints[http_post_uri_file_index], response_uri_len);
     } else {
-        snprintf(response_uri, response_uri_len, "/%ld", status_code);
+        snprintf(response_uri, response_uri_len, "/%" PRIu32, status_code);
```
Comment (DRracer) [inline lib/WUI/http/httpd_post.c:357]: this is a nice example of formatting macros, please report to the rest of the team :+1:

File: tests/stubs/cmsis_os.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (DRracer) [inline tests/stubs/cmsis_os.h:1]: Do you think this stub is usable for other than unit-tests? You propose not to place it into the `tests/unit` directory subtree.

File: lib/WUI/http/httpd_post.c (hunk 1 above)
Comment (vorner) [reply lib/WUI/http/httpd_post.c:357]: Report where? Meeting? Some channel? Yes, I'd like to set up some kind of quality guild, but I haven't gotten around that yet.

File: tests/stubs/cmsis_os.h (hunk 2 above)
Comment (vorner) [reply tests/stubs/cmsis_os.h:1]: It's the nearest `stubs` directory I've found. I could create `tests/unit/stubs`… on the other hand, once we have the fuzzers, they are not exactly unit tests and they would probably like to reuse it.

Comment (DRracer) [reply tests/stubs/cmsis_os.h:1]: ok, fair enough

**Review by DRracer — APPROVED** [review]

**Pull request merged**